### PR TITLE
Feature: contract-loader-fs

### PR DESCRIPTION
**Pull request by chmanie**
This PR adds a contract loader for the filesystem to use in node.js. This is part of the effort to support loaders in our reputation-mining-client.

### CODE REVIEW

File: packages/colony-js-contract-loader-fs/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+## Installation
+
+```
+yarn add @colony/colony-js-contract-loader-fs
+```
+
+
+## Usage
+
+// TODO DOCS
```
Comment (JamesLefrere) [inline packages/colony-js-contract-loader-fs/README.md]: Seems fine to me so long as we remember to come back here.

File: packages/colony-js-contract-loader-fs/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# colonyJS Contract Loader for Node.js using the filesystem
```
Comment (JamesLefrere) [inline packages/colony-js-contract-loader-fs/README.md]: 🙌

File: packages/colony-js-contract-loader-fs/src/loaders/FSLoader.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  Query,
+} from '@colony/colony-js-contract-loader';
+
+import type { ConstructorArgs } from '../flowtypes';
+
+const assert = require('assert');
+const path = require('path');
+const jsonfile = require('jsonfile');
+
+const ContractLoader = require('@colony/colony-js-contract-loader').default;
```
Comment (JamesLefrere) [inline packages/colony-js-contract-loader-fs/src/loaders/FSLoader.js]: Why require over import? I'm surprised eslint doesn't complain about this.

File: packages/colony-js-contract-loader-fs/src/loaders/FSLoader.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  ): Promise<?ContractDefinition> {
+    const { contractName = '' } = query;
+
+    assert(!!contractName, 'A `contractName` option must be provided');
+
+    const file = path.resolve(this._contractDir, `${contractName}.json`);
+    return new Promise((resolve, reject) => {
+      jsonfile.readFile(file, (err, contents) => {
+        if (err) return reject(err);
+        const transformed = this._transform(contents, query, requiredProps);
```
Comment (JamesLefrere) [inline packages/colony-js-contract-loader-fs/src/loaders/FSLoader.js]: What if there's an error in `this._transform`? You might need try/catch -> reject.

File: packages/colony-js-contract-loader/ContractLoader.js [hunk 5]
```diff
@@ -1,40 +1,35 @@
-/* @flow */
-
-import assert from 'browser-assert';
-import 'isomorphic-fetch';
+const assert = require('assert');
```
Comment (JamesLefrere) [inline packages/colony-js-contract-loader/ContractLoader.js]: `browser-assert` is still used elsewhere in the project. Can it be replaced?

File: packages/colony-js-contract-loader/ContractLoader.js [hunk 6]
```diff
@@ -161,7 +95,7 @@ export default class ContractHttpLoader implements IContractLoader {
         ...otherQuery,
       },
       requiredProps,
-    );
+    ) || {};
```
Comment (JamesLefrere) [inline packages/colony-js-contract-loader/ContractLoader.js]: 👍

File: packages/colony-js-contract-loader-fs/src/loaders/TruffleLoader.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
+/* @flow */
```
Comment (JamesLefrere) [inline packages/colony-js-contract-loader-fs/src/loaders/TruffleLoader.js]: Have a look at putting this in `@colony/colony-js-contract-loader` (and do the same for `TrufflepigLoader`.

File: packages/colony-js-contract-loader-fs/src/loaders/FSLoader.js (hunk 3 above)
Comment (chmanie) [reply packages/colony-js-contract-loader-fs/src/loaders/FSLoader.js]: Eslint is totally fine with `require` as it always is a viable option :) This package is supposed to be run in node only so I ditched all the babel business and went with removing the flow types only on build. Thus `require` and not `import`

File: packages/colony-js-contract-loader/ContractLoader.js (hunk 5 above)
Comment (chmanie) [reply packages/colony-js-contract-loader/ContractLoader.js]: I can do that!

File: packages/colony-js-contract-loader-fs/src/loaders/TruffleLoader.js (hunk 7 above)
Comment (chmanie) [reply packages/colony-js-contract-loader-fs/src/loaders/TruffleLoader.js]: On it!

File: packages/colony-js-contract-loader-fs/src/loaders/FSLoader.js (hunk 3 above)
Comment (JamesLefrere) [reply packages/colony-js-contract-loader-fs/src/loaders/FSLoader.js]: Ok that makes sense 👍

**Pull request merged**